Source/JavaScriptCore/yarr/YarrJIT.cpp

```diff
@@ -2051,11 +2051,17 @@ class YarrGenerator final : public YarrJITInfo {
             unsigned currPosition = currTerm->inputPosition;
 
             if (currPosition > lastPosition) {
+                // If the next term is too far away, we'll handle it by itself
+                if (currPosition > lastPosition + 16)
+                    break;
                 if (currPosition > lastPosition + 1)
                     opList.insertFill(lastPosition - firstPosition + 1, nullptr, currPosition - lastPosition - 1);
                 opList.append(currOp);
                 lastPosition = currPosition;
             } else if (currPosition < firstPosition) {
+                // If the next term is too far away, we'll handle it by itself
+                if (currPosition < firstPosition - 16)
+                    break;
```

Comment:
Can we have this `16` as a readable constant like, `constexpr xxx = 16;` and use this?